feat(nfs-server): Add NodeAffinityList override to StorageClass

Rid opened the pull request:
Signed-off-by: Rid <rid@cylo.io>

**Why is this PR required? What issue does it fix?**:
Resolves #146 
This PR allows a user to add a node affinity list to the `cas.openebs.io/config` configmap in the `StorageClass`, which can override the `OPENEBS_IO_NFS_SERVER_NODE_AFFINITY` environment variable.

**What this PR does?**:
This PR allows users to specify affinity via the `StorageClass`, giving the user control of what nodes run the NFS server in a more dynamic way.

Sample `StorageClass`:
```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: my-new-storageclass
  namespace: default
  annotations:
    openebs.io/cas-type: nfsrwx
    cas.openebs.io/config: |
      - name: NFSServerType
        value: "kernel"
      - name: BackendStorageClass
        value: "replicated-storage"
      - name: NodeAffinityLabels
        list:
          - node1
          - node2
provisioner: openebs.io/nfsrwx
reclaimPolicy: Delete
```

**Does this PR require any upgrade changes?**:
No.

**Checklist:**
- [x] Fixes #146 
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated? 
- [x] Commit has unit tests
- [x] Commit has integration tests

Rid commented:
@niladrih @mynktl PTAL

niladrih commented:
Thank you for your contribution @Rid. Canonically we have a similar config parameter in the openebs/dynamic-localpv-provisioner project. Could you adapt your implementation to follow the same syntax in the StorageClass?

https://github.com/openebs/dynamic-localpv-provisioner/search?q=NodeAffinityLabels

Also, the cas-template library accepts lists with the list key. That might be better suited for the list: https://github.com/niladrih/maya/blob/v2.12.x/pkg/apis/openebs.io/v1alpha1/cas_template.go#L82

Rid committed:
feat(nfs-server): Add NodeAffinityList override to StorageClass

Signed-off-by: Rid <rid@cylo.io>

Rid commented:
@niladrih Thanks for the info, I've changed the PR to use the list method and updated the nomenclature to the same as `openebs/dynamic-localpv-provisioner`.

niladrih commented:
Thanks. Before I review, I must let you know that the node affinity label keys are used in cases when the kubernetes.io/hostname label's values aren't unique across nodes. These do not help in scheduling the NFS server or the application Pod in any way. 

For scheduling, you'd want to try using the AllowedTopologies parameter in the storageclass.
https://kubernetes.io/docs/concepts/storage/storage-classes/#allowed-topologies

To configure scheduling for the NFS server, you'd want to add the AllowedTopologies parameter to the backend storageclass.

To configure scheduling for the application Pod, you'd want to add the AllowedTopologies parameter to the NFS StorageClass.

Rid commented:
@niladrih Thanks, you're right. What I want can be achieved via `AllowedTopologies` on the `StorageClass`. Should have read the docs first!
